wa/commands/process.py

```diff
@@ -0,0 +1,84 @@
+#    Copyright 2014-2018 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import sys
+
+from wa import Command, settings
+from wa import discover_wa_outputs
+from wa.framework import pluginloader
+from wa.framework.version import get_wa_version
+from wa.framework.output_processor import OutputProcessor
+from wa.utils.log import log_error
+
+class ProcessCommand(Command):
+
+    name = 'process'
+    description = 'Process the output from previously run workloads.'
+
+    def __init__(self, subparsers):
+        self.loader = pluginloader
+        super(ProcessCommand, self).__init__(subparsers)
+
+    def initialize(self, context):
+        self.parser.add_argument('-d', '--process-directory', metavar='DIR', default=None,
+                                 help="""
+                                 Specify a directory to walk to find the data
+                                 from previous runs to process .If this option is
+                                 not specified, then {} will be used instead.
+                                 """.format(settings.default_output_directory))
+
+    def execute(self, config, args):
+        self.logger.debug('Version: {}'.format(get_wa_version()))
+        self.logger.debug('Command Line: {}'.format(' '.join(sys.argv)))
+
+        if args.process_directory:
+            process_directory = args.process_directory
+        else:
+            process_directory = settings.default_output_directory
+        process_directory = os.path.expandvars(process_directory)
+        self.logger.debug('Using process directory: {}'.format(process_directory))
+        if not os.path.exists(process_directory):
+            self.logger.error('Path "{}" does not exist'.format(process_directory))
+            return
+        for run_output in discover_wa_outputs(process_directory):
```

Comment:
Don't do this, at least not by default. The directory should be a WA output directory not contain it. Use `RunOutput` directly here instead.

If you want to add the ability to recursively process a directory tree with multiple outputs, make the user explicitly specify that that is their intent, e.g. with `-r` opition.